liquidator/liquidator.js

```diff
@@ -284,8 +286,30 @@ class Liquidator {
 
       // Send the transaction or report failure.
       let receipt;
+
       try {
-        receipt = await liquidation.send(txnConfig);
+        // Make sure to keep trying with this nonce
+        const nonce = await this.web3.eth.getTransactionCount(this.account);
+
+        // Min Gas Price, with a max gasPrice of double
+        const minGasPrice = parseInt(this.gasEstimator.getCurrentFastPrice(), 10);
+        const maxGasPrice = 2 * minGasPrice;
+
+        // Increment by 10 gwei per trial
+        const gasPriceScalingFunction = ynatm.LINEAR(10);
+
+        // Receipt without events
+        receipt = await ynatm.send({
```

Comment:
if theres a chance liquidation could fail, ie revert, make sure this properly stops retrying. i saw you have a way to do that now in your library

Reply:
By default, it will now immediately return and throw an error if the tx receipt contains a "revert" string

Reply:
that is quite opinionated of you, but i think it works 😄

Reply:
You can override that tho! 🤩

Reply:
Perhaps this was made clear somewhere else, but how many times will this call `liquidation.send()` before stopping?

> it will now immediately return and throw an error if the tx receipt contains a "revert" string
Where is this logic?

Reply:
Thats dependent on the chosen `gasPriceScalingFunction` for example a linear scaling function that increments it by 1, with a minGasPrice of 5, and a maxGasPrice of 15, delay of 10000(ms) will result in 10 attempted calls, with a minute delay between each of them.

Reply:
Ok that's a great feature. I think its reasonable to try to send it 3 times, increasing gas 25 each time? WDYT @kendricktan @chrismaree @mrice32 @daywiss

Reply:
Forgot to mention that if one of the tx gets mined then it immediately returns too

Reply:
Is that enough of a ramp up though @nicholaspai ? the gas prices on black Thursday were in the 500 Gwei range and bots were sending out transactions well over 1000 Gwei. If we started at 500 and incremented by 25 each time it's not really going to help the situation. I think there should be a doubling action in the ramp up perhaps?

Reply:
Good thinking, agreed with doubling the ramp up. I think its ok to pay high gas prices because its important that these transactions are ultimately mined in a reasonable time.

Reply:
@kendricktan can you implement the curve that doubles every txn? Up to a max 3 times?

Reply:
@kendricktan where does it define here that it has 3 re-tries?

Reply:
@kendricktan?